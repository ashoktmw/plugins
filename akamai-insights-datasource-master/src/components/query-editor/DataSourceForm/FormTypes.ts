import { SelectableValue } from '@grafana/data';
import { groupBy } from 'lodash';

import { enumToSelectableValues, prettyEnum, stringsToSelectableValues } from '../../../utils/utils';

export const shortLabelWidth = 25;
export const mediumLabelWidth = 50;

export enum FilterType {
  Dimension = 'Dimension',
  Metric = 'Metric'
}

export const filterTypeOptions = enumToSelectableValues(FilterType);

export const filterTypeOptionsMap = groupBy(filterTypeOptions, 'value');

export interface FilterFormModel {
  type: SelectableValue<string>;
  query: FilterQueryFormModel;
}

export interface FilterQueryFormModel {
  name?: SelectableValue<string>;
  operator?: SelectableValue<string>;
  expressions?: string[];
}

export enum DimensionOperator {
  InList = 'IN_LIST',
  MatchRegExp = 'MATCH_REGEXP',
  NotMatchRegExp = 'NOT_MATCH_REGEXP',
  BeginsWith = 'BEGINS_WITH',
  NotBeginsWith = 'NOT_BEGINS_WITH',
  EndsWith = 'ENDS_WITH',
  NotEndsWith = 'NOT_ENDS_WITH',
  Contains = 'CONTAINS',
  NotContains = 'NOT_CONTAINS',
  Exact = 'EXACT'
}

export enum FilterValueType {
  Text = 'TEXT',
  Enum = 'ENUM'
}

export const dimensionOperatorOptions = enumToSelectableValues(DimensionOperator);

export const authorizableOrEnumFiltersOperatorOptions = stringsToSelectableValues([ DimensionOperator.InList ], prettyEnum);

export enum MetricOperator {
  Equal = 'EQUAL',
  NotEqual = 'NOT_EQUAL',
  LessThan = 'LESS_THAN',
  GreaterThan = 'GREATER_THAN'
}

export const metricOperatorOptions = enumToSelectableValues(MetricOperator);

export interface SortByFormModel {
  type?: SelectableValue<string>;
  query: SortByQueryFormModel;
}

export interface SortByQueryFormModel {
  name?: SelectableValue<string>;
  sortOrder?: SelectableValue<string>;
}

export enum SortByOrder {
  Ascending = 'ASCENDING',
  Descending = 'DESCENDING'
}

export const sortByOrderOptions = enumToSelectableValues(SortByOrder);
