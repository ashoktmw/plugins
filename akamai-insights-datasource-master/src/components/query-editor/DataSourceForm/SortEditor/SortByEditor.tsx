import { SelectableValue } from '@grafana/data';
import { InlineField, Select } from '@grafana/ui';
import React, { Fragment, JSX } from 'react';

import { shortLabelWidth, sortByOrderOptions, SortByQueryFormModel } from '../FormTypes';

export interface SortByEditorProps {
  model: SortByQueryFormModel;
  names: SelectableValue<string>[];
  onChange: (query: SortByQueryFormModel) => void;
}

export function SortByEditor({ model, names, onChange }: SortByEditorProps): JSX.Element {
  return (
    <Fragment>
      <InlineField>
        <Select
          placeholder="Name"
          isClearable={false}
          width={shortLabelWidth}
          options={names}
          value={model.name}
          onChange={name => onChange({ ...model, name })}
        />
      </InlineField>

      <InlineField>
        <Select
          placeholder="Sort order"
          isClearable={false}
          width={shortLabelWidth}
          options={sortByOrderOptions}
          value={model.sortOrder}
          onChange={sortOrder => onChange({ ...model, sortOrder } )}
        />
      </InlineField>
    </Fragment>
  );
}
