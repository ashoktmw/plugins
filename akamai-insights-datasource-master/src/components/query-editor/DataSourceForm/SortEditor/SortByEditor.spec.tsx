import React from 'react';
import renderer from 'react-test-renderer';

import { SortByEditor } from './SortByEditor';
import { mockDimensionsOptions, mockMetricsOptions } from '../../../../test/mocks/mock-selectable-values';
import { sortByOrderOptions, SortByQueryFormModel } from '../FormTypes';

describe('SortEditor component', () => {
  it('should render dimension editor', () => {
    const model: SortByQueryFormModel = {
      name: mockDimensionsOptions[ 1 ],
      sortOrder: sortByOrderOptions[ 1 ]
    };

    const names = mockDimensionsOptions;
    const onChange = jest.fn();

    const tree = renderer.create(
      <SortByEditor model={model} names={names} onChange={onChange}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render metric editor', () => {
    const model: SortByQueryFormModel = {
      name: mockMetricsOptions[ 1 ],
      sortOrder: sortByOrderOptions[ 1 ]
    };

    const names = mockMetricsOptions;
    const onChange = jest.fn();

    const tree = renderer.create(
      <SortByEditor model={model} names={names} onChange={onChange}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
