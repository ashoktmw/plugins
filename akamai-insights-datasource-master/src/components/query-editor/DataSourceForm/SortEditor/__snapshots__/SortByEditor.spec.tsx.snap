// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortEditor component should render dimension editor 1`] = `
[
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              time1Hour
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              Descending
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`SortEditor component should render metric editor 1`] = `
[
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              edgeHitsSum
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-4-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-1f5gueu"
  >
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-rmjxhr-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-5-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1kl463j-grafana-select-value-container"
          >
            <div
              className="css-18bh10p-singleValue css-upz218-SingleValue"
            >
              Descending
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-5-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
